global.php

```diff
@@ -224,18 +241,37 @@
 }
 
 // Fetch the theme to load from the cache
+if($loadstyle != "def='1'")
+{
+	$query = $db->simple_select('themes', 'name, tid, properties, stylesheets, allowedgroups', $loadstyle, array('limit' => 1));
+	$theme = $db->fetch_array($query);
+
+	if(isset($theme['tid']) && !$load_from_forum && !is_member($theme['allowedgroups']) && $theme['allowedgroups'] != 'all')
+	{
+		if($load_from_user == 1)
+		{
+			$db->update_query('users', array('style' => 0), "style = '{$mybb->user['style']}'");
+		}
+
+		if(isset($mybb->cookies['mybbtheme']))
+		{
+			my_unsetcookie('mybbtheme');
+		}
+
+		$loadstyle = "def='1'";
+	}
+}
+
 if($loadstyle == "def='1'")
 {
 	if(!$cache->read('default_theme'))
 	{
 		$cache->update_default_theme();
 	}
-	$theme = $cache->read('default_theme');
-}
-else
-{
-	$query = $db->simple_select('themes', 'name, tid, properties, stylesheets', $loadstyle, array('limit' => 1));
-	$theme = $db->fetch_array($query);
+
+	$theme = $cache->read('default_theme', true);
```

Comment:
Any reason why you added the second parameter? That'd mean that the database cache gets useless.